cpp/src/arrow/compute/exec/expression_test.cc

```diff
@@ -522,7 +522,7 @@ TEST(Expression, BindNestedCall) {
 
 TEST(Expression, ExecuteFieldRef) {
   auto ExpectRefIs = [](FieldRef ref, Datum in, Datum expected) {
-    auto expr = field_ref(ref);
+    auto expr = field_ref(std::move(ref));
```

Comment:
This is fine, but I think it's acceptable to be sloppier with moves in tests because it makes debugging easier